### DISCUSSION TITLE
Description of FourCC's used in the GPMF header metadata

### DISCUSSION THREAD

**Discussion by TobyHaddon** [gopro/labs#711] [Q&A]
Can anyone point me to descriptions for the FourCCs used in GoPro generated MP4 header metadata. Many thanks Toby

FMWR
LINF
CINF
CASN
MINF
MUID
CMOD
MTYP
OREN
DZOM
DZST
SMTR
PRTN
PTWB
PTSH
PTCL
EXPT
PIMX
PIMN
PTEV
RATE
SROT
EISE
EISA
HCTL
AUPT
APTO
AUDO
BROD
BRID
PVUL
PRJT
SOFF
CLKS
CDAT
SCTM
PRNA
PRNU
SCAP
CDTM
DUST
VRES
VFPS
HSGT
BITR
MMOD
RAMP
TZON
CLKC
DZMX
CTRL
PWPR
ORDP
CLDP
PIMD
DVID
DVNM
ABSC
ZFOV
VFOV
MXCF
MAPX
MYCF
MAPY
PYCF
POLY
ZMPL
ARUW
ARWA

**Comment by dnewman-gpsw** [comment]
The is no public list to my knowledge. Some might be on https://github.com/gopro/gpmf-parser. A few are obvious like FMWR  and PTWB, some are likely internal only. Others like the last dozen listed are really hard to explain.  I don't have time to document them all, plus I only know the few I use.  Is there a particular piece of info you seek?

**Comment by TobyHaddon** [comment]
I’m writing software to process stereo GoPro cameras used for measuring fish underwater.  I’m looking at ways to reduced errors from surveys and to help automate the processing of the information.  I’m interested in the header metadata to check if both cameras were set-up correctly and both have the same configuration.  Let me adjust settings and see the changes in the header metadata and if I have questions I'll repost. Thanks

**Comment by TobyHaddon** [comment]
Is there a way to extract the GPMF header metadata programmatically (as opposed to the streaming metadata)?  Is the code that drives https://gopro.github.io/labs/control/metadata/ available?

**Reply by dnewman-gpsw** [reply]
That link is to source code in javascript, you can use and/or convert that.

**Comment by TobyHaddon** [comment]
Thanks
